P0009/P0009.md

```diff
@@ -1407,26 +1418,39 @@ template<size_t... OtherExtents>
 
 ```c++
 template<class... SizeTypes>
-  explicit constexpr extents(SizeTypes... dynamic) noexcept;
+  explicit constexpr extents(SizeTypes... exts) noexcept;
 ```
 
-* [4]{.pnum} *Constraints:*
+* [5]{.pnum} *Constraints:*
     
-    * [4.1]{.pnum} `(is_convertible_v<SizeTypes, size_type> && ...)` is `true`, and
+    * [5.1]{.pnum} `(is_convertible_v<SizeTypes, size_type> && ...)` is `true`, and
     
-    * [4.2]{.pnum} `sizeof...(SizeTypes) == rank_dynamic()` is `true`.
+    * [5.2]{.pnum} `sizeof...(SizeTypes)` equals either `dynamic_rank()` or `rank()`.
```

Comment:
Why do not allow any value in range [dynamic_rank(), rank()). Any further extends would need to be equal to the corresponding static extend. But this is already the case for any static rank, present before last dynamic extend.

Reply:
This was answered by above.

Reply:
This was the majority vote from the authors, only allowing either `rank_dynamic` or `rank` (just noticed I have a word swap there for `rank_dynamic`). If we did something different I think it had to be between "arguments up to last dynamic rank" and rank. It wouldn't be between "rank_dynamic" and "rank" in that case. That said I did do a test implementation of all options.

Reply:
The constructor of `mdspan` (and `extends`) should be conditionally explicit if `sizeof...(Args) > rank_dynamic()` or `N > rank_dynamic()`, i.e. when there is possibility of UB, to match span (ptr, size) constructor.
Note that `N > rank_dynamic()` is same as `N == rank() && rank_dynamic() != rank()` with your precondition. Could be also spelled as `N != rank_dynamic()`

Reply:
yeah it should be. Doing that now.

Reply:
Thanks for looking at this!